stress.yaml

```diff
@@ -27,19 +27,32 @@
     - set_fact: index={{ec2_tag_index}}
       when: index is not defined
 
+    - set_fact: taskset="taskset -c {{load_cpus}}"
+      when: load_cpus is defined
+
+    - set_fact: taskset=""
+      when: load_cpus is not defined
+
+    - debug: msg="setting network IRQ affinity to CPU0"
+      when: load_cpus is defined
+
+    - shell: for irq in `cat  /proc/interrupts | grep eth0 | cut -d":" -f1`; do echo 1 > /proc/irq/$irq/smp_affinity; done
+      when: load_cpus is defined
+      sudo: yes
+
     - set_fact: log_file="{{remote_path}}{{output_file}}.{{ec2_region}}.{{index}}"
     - set_fact: range_start={{1 + (populate|int) * (index|int) }}
     - set_fact: range_end={{(populate|int) * (1 + (index|int)) }}
     - set_fact: range="{{range_start}}..{{range_end}}"
     - debug: msg="client key range {{range}}"
```

Comment:
2015-11-05 7:59 GMT+01:00 Tzach Livyatan <notifications@github.com>:

> In stress.yaml
> <https://github.com/scylladb/cassandra-test-and-deploy/pull/21#discussion_r43981507>
> :
>
> > @@ -27,19 +27,32 @@
> >      - set_fact: index={{ec2_tag_index}}
> >        when: index is not defined
> >
> > +    - set_fact: taskset="taskset -c {{load_cpus}}"
> > +      when: load_cpus is defined
> > +
> > +    - set_fact: taskset=""
> > +      when: load_cpus is not defined
> > +
> > +    - debug: msg="setting network IRQ affinity to CPU0"
> > +      when: load_cpus is defined
> > +
> > +    - shell: for irq in `cat  /proc/interrupts | grep eth0 | cut -d":" -f1`; do echo 1 > /proc/irq/$irq/smp_affinity; done
> > +      when: load_cpus is defined
> > +      sudo: yes
> > +
> >      - set_fact: log_file="{{remote_path}}{{output_file}}.{{ec2_region}}.{{index}}"
> >      - set_fact: range_start={{1 + (populate|int) * (index|int) }}
> >      - set_fact: range_end={{(populate|int) * (1 + (index|int)) }}
> >      - set_fact: range="{{range_start}}..{{range_end}}"
> >      - debug: msg="client key range {{range}}"
>
> @glommer <https://github.com/glommer> as far as you know, do we need to
> set different population per client?
> @tgrabiec <https://github.com/tgrabiec> did you do so in yours
> Scylla/Cassandra benchmarks?
>

In my benchmarks I used the same population for all clients. It depends
what kind of workload you want to test.